pkg/cloudprovider/gcp/volume_snapshotter.go

```diff
@@ -60,25 +58,22 @@ func (b *VolumeSnapshotter) Init(config map[string]string) error {
 		return err
 	}
 
-	b.snapshotLocation = config[snapshotLocationKey]
-
-	project, err := extractProjectFromCreds()
+	/* Works with both credential files and the default compute engine service account */
+	creds, err := google.FindDefaultCredentials(oauth2.NoContext, compute.ComputeScope)
```

Comment:
If I'm not mistaken, this will error if `GOOGLE_APPLICATION_CREDENTIALS` has a value, but no file is found at that path. By default, `velero install` is going to set this env var to a value of `/credentials/cloud`.  I think we probably need to change the `velero install` logic so if `--no-secret` is specified, these env vars aren't set. @nrb does that make sense to you?

Reply:
I checked the velero install logic, and it doesn't create a file if --no-secret is specified. I also tested it using --no-secret, and it seemes to initialize properly using the workload identity.

Reply:
OK -- I'll do some testing as well.

Reply:
Yep, you're right - specifying `--no-secret` removes the env vars entirely from the deployment. So we're good here.